feat(aio): set the document title in the browser title

petebacondarwin opened the pull request.

mary-poppins commented:
The angular.io preview for e81f63e65bfe65a190c87025cfdc990b4ace3bb3 is available [here][1].

[1]: https://pr16031-e81f63e65bfe65a190c87025cfdc990b4ace3bb3.ngbuilds.io/

Foxandxss reviewed (changes requested):
Other things than docs gives undefined.

petebacondarwin commented:
Good point. I'll fix it up

petebacondarwin commented:
By ensuring `doc.title` is never undefined

mary-poppins commented:
The angular.io preview for 3f51460462268f5ecdb55960352b094ad6f637ae is available [here][1].

[1]: https://pr16031-3f51460462268f5ecdb55960352b094ad6f637ae.ngbuilds.io/

petebacondarwin committed:
feat(aio): set the document title in the browser title

petebacondarwin committed:
build(aio): extract the title from the content if necessary

Documents can specify their title via the `title` or `name` jsdoc tags.

This change adds that, if neither are provided, the first `<h1>` element
is removed from the `renderedContent` and used for the title.

If there is still no title then it is set to the empty string and a warning
is logged.

mary-poppins commented:
The angular.io preview for d35abaa468ef2407ae6d2c67d96cdddd9753014d is available [here][1].

[1]: https://pr16031-d35abaa468ef2407ae6d2c67d96cdddd9753014d.ngbuilds.io/

wardbell reviewed (approved):
LGTM

gkalpak commented on aio/tools/transforms/angular.io-package/processors/convertToJson.js:
```diff
@@ -8,13 +8,27 @@ module.exports = function convertToJsonProcessor() {
+
+          // If there is no title then try to extract it from the first h1 in the renderedContent
+          if (title === undefined) {
+            contents = contents.replace(/<h1[^>]*>(.+?)<\/h1>/, (outerHeading, innerHeading) => {
```
Why `replace` when you are not actually replacing it? I assume it is a left-over from a previous implementation that _did_ replace it.

petebacondarwin replied:
you're right. I'll change that

petebacondarwin commented:
@gkalpak PTAL

gkalpak commented on aio/tools/transforms/angular.io-package/processors/convertToJson.spec.js:
```diff
@@ -0,0 +1,55 @@
+  });
+
+  it('should log a warning', () => {
+    const docs = [ { docType: 'test-doc' }];
```
Consistently inconsistent spacing (`[ {` vs `}]`).
<sub>(Which is way better than inconsistently inconsistent spacing, but still... 😛 </sub>

petebacondarwin replied:
:-(

gkalpak reviewed (approved):
LGTM

mary-poppins commented:
The angular.io preview for e83c09e964ff584f66ae08562e7999792d9a2b88 is available [here][1].

[1]: https://pr16031-e83c09e964ff584f66ae08562e7999792d9a2b88.ngbuilds.io/
